src/compiler/transformers/declarations/emitResolver.ts

```diff
@@ -122,13 +118,39 @@ export function createEmitDeclarationResolver(file: SourceFile): IsolatedEmitRes
         }
         return undefined;
     }
+
+    function resolveEntityName(location: Node, node: Expression, meaning: SymbolFlags): EmitDeclarationSymbol | undefined {

```

Comment:
Shouldn't this be `resolveEntityName(location: EnumDeclaration, ...` ?
And on the same point, I feel using a specific name would make more sense (e.g. enumDecl).